tests/test_hep_bd0xx.py

```diff
@@ -598,6 +598,38 @@ def test_texkeys_from_035__z_9_and_035__a_9():
     assert expected == result['035']
 
 
+def test_desy_bookkeekping_from_035__z_9_DESY():
+    schema = load_schema('hep')
+    subschema = schema['properties']['_desy_bookkeeping']
+
+    snippet = (
+        '<datafield tag="035" ind1=" " ind2=" ">'
+        '  <subfield code="9">DESY</subfield>'
+        '  <subfield code="z">DA17-kp47ch</subfield>'
+        '</datafield>'
+    )  # record/1635310
+
+    expected = [
+        {
+            'identifier': 'DA17-kp47ch',
+        },
+    ]
+    result = hep.do(create_record(snippet))
+
+    assert validate(result['_desy_bookkeeping'], subschema) is None
+    assert expected == result['_desy_bookkeeping']
+
+    expected = [
+        {
+            '9': 'DESY',
+            'z': 'DA17-kp47ch',
+        },
+    ]
+    result = hep2marc.do(result)
+
+    assert expected == result['035']
```

Comment:
I would add a test with both `035` and `595_D` in the snippet to ensure that the two rules do not clobber each other's output.